code/datums/status_effects/debuffs/fire_stacks.dm

```diff
@@ -0,0 +1,261 @@
+/datum/status_effect/fire_handler
+	duration = -1
+	alert_type = null
+	status_type = STATUS_EFFECT_REFRESH //Custom code
+	on_remove_on_mob_delete = TRUE
+	tick_interval = 2 SECONDS
+	/// Current amount of stacks we have
+	var/stacks
+	/// Maximum of stacks that we could possibly get
+	var/stack_limit = 20
+	/// What status effect types do we remove uppon being applied. These are just deleted without any deduction from our or their stacks when forced.
+	var/list/enemy_types
+	/// What status effect types do we merge into if they exist. Ignored when forced.
+	var/list/merge_types
+	/// What status effect types do we override if they exist. These are simply deleted when forced.
+	var/list/override_types
+	/// For how much firestacks does one our stack count
+	var/stack_modifier = 1
+
+/datum/status_effect/fire_handler/refresh(mob/living/new_owner, new_stacks, forced = FALSE)
+	if(forced)
+		set_stacks(new_stacks)
+	else
+		adjust_stacks(new_stacks)
+
+/datum/status_effect/fire_handler/on_creation(mob/living/new_owner, new_stacks, forced = FALSE)
+	if(isanimal(owner))
+		qdel(src)
+		return
+
+	owner = new_owner
+	set_stacks(new_stacks)
+
+	for(var/enemy_type in enemy_types)
+		var/datum/status_effect/fire_handler/enemy_effect = owner.has_status_effect(enemy_type)
+		if(enemy_effect)
+			if(forced)
+				qdel(enemy_effect)
+				continue
+
+			var/cur_stacks = stacks
+			adjust_stacks(-enemy_effect.stacks * enemy_effect.stack_modifier / stack_modifier)
+			enemy_effect.adjust_stacks(-cur_stacks * stack_modifier / enemy_effect.stack_modifier)
+			if(enemy_effect.stacks <= 0)
+				qdel(enemy_effect)
+
+			if(stacks <= 0)
+				qdel(src)
+				return
+
+	if(!forced)
+		var/list/merge_effects = list()
+		for(var/merge_type in merge_types)
+			var/datum/status_effect/fire_handler/merge_effect = owner.has_status_effect(merge_type)
+			if(merge_effect)
+				merge_effects += merge_effects
+
+		if(LAZYLEN(merge_effects))
+			for(var/datum/status_effect/fire_handler/merge_effect in merge_effects)
+				merge_effect.adjust_stacks(stacks * stack_modifier / merge_effect.stack_modifier / LAZYLEN(merge_effects))
+			qdel(src)
+			return
+
+	for(var/override_type in override_types)
+		var/datum/status_effect/fire_handler/override_effect = owner.has_status_effect(override_type)
+		if(override_effect)
+			if(forced)
+				qdel(override_effect)
+				continue
+
+			adjust_stacks(override_effect.stacks)
+			qdel(override_effect)
+
+	. = ..()
+	cache_stacks()
```

Comment:
I am still kinda miffed that this is required. I don't think the cost of calling parent here earlier is that high, and it would remove a class of error from people working with your system. Is there a strong reason to not do it?